docs/contributing/core/continuous-integration.md

```diff
@@ -0,0 +1,111 @@
+---
+myst:
+  html_meta:
+    "description": "Essential continuous integration practices"
+    "property=og:description": "Essential continuous integration practices"
+    "property=og:title": "Essential continuous integration practices"
+    "keywords": "Plone, continuous integration, best practices"
+---
+
+# Essential continuous integration practices
```

Comment:
This whole page is outdated, it does not mention `mr.roboto`, GHA nor `plone/meta` which make almost all the rules described outdated.

Sorry that it took me forever to react on the call to help on this PR 🙇🏾 

I can do the update, should I push it directly to this branch? 🤔

Reply:
@gforcada please create a PR against this branch for review. Do not push commits directly to this branch.

I'm not clear on which repositories `plone/meta` applies. From what I know, many repositories use `make` and not `tox`, including Volto, plone.restapi, and Documentation. Until there is both agreement and implementation for a unified approach, I don't know whether `plone/meta` should be declared as authoritative for all projects under the Plone GitHub organization.

Reply:
Ok, thanks I will do that (the branch).

As for `plone/meta` or `make`, well, it's literally all but volto related packages that already use `plone/meta`. There are a few that are missing (like CMFPlone) that I did not had the time to apply `plone/meta` to it.

But this documentation is meant for classic UI, or? 🤔 I meant I saw quite a few references that always branch to say "if you are doing volto then see ../volto"

Reply:
This part of the  documentation is intended for anything that is not Volto, which includes Classic UI, core add-ons, and other core packages.

There is also an effort with cookiecutter and mxdev, which is intended to replace buildout, at which point another rewrite of docs will need to take place.

I don't know how that reconciles with `plone/meta`. I just document what I understand.

Reply:
`plone/meta` and `mxdev`/`cookiecutter` are orthogonal, at least I'm 100% sure with `mxdev`, with `cookiecutter` that depends on what's being done, of for which purpose is used 😓 

I've seen so many `cookieXX` repositories, that I'm no longer sure what one should use 🤷🏾 

but if the goal is to replace `zc.buildout`, then `plone/meta` has nothing to do with that and should be fine to document.

Actually `plone/meta` has quite some exhaustive documentation on its own 😄

Reply:
@gforcada even before your review and subsequent pull request, I had a lot of issues with this document. It never had a high-level review. I apologize for not bringing up my concerns earlier, but here they are now.

1. It's not clear what is the purpose of this document.
2. It's not clear who is the target audience.
3. Is it a tutorial, a how-to guide, an explanation, or a reference? See https://diataxis.fr/ for details of what these categories mean.
4. Much of the content is redundant to existing parts of the documentation, as well as within this pull request, including:
   - https://6.docs.plone.org/contributing/first-time.html#set-up-your-environment
   - https://6.docs.plone.org/contributing/index.html
   - https://github.com/plone/documentation/pull/1671/files#diff-92572efaca37b0041952183745916a031065e6fc8f04a0a5faa26ad023dc609f

Once we have an idea of what we want to do, then we can rewrite this document to achieve that goal. I'd be happy to have a chat with you in real-time in Discord, or continue the conversation here.

----

For cookie*, the primary repo is https://github.com/plone/cookieplone. It uses templates found in https://github.com/plone/cookieplone-templates. I don't know whether that repo uses `plone/meta` itself, but [its templates do](https://github.com/search?q=repo%3Aplone%2Fcookieplone-templates%20meta&type=code).

cookie* uses mxdev, as described in https://6.docs.plone.org/install/manage-add-ons-packages.html#manage-plone-backend-packages-with-mxdev.

----

Are these files `plone/meta`'s documentation?

- https://github.com/plone/meta/blob/main/README.md
- https://github.com/plone/meta/blob/main/config/README.md

Reply:
In reverse order 🙃 

- yes, that's the documentation for `plone/meta`

- oh nice, thanks for the cookie pointers, now it's a bit more clear to me, the (not to be answered and solved now) is how that works with mr.bob and bobtemplates.plone and all that ecosystem, or again, we have cookie* for volto and mr.bob for classic UI? 🤔 

- for this document, I would say that it's an explanation of how the CI system works, being the target audience newcomers.

With the links you provided (thanks!) I would say it could be linked from https://6.docs.plone.org/contributing/index.html#continuous-integration, then, we could simplify my PR to remove the explanation about PRs and be more focused only on the tooling involved.

How that sounds?

As for talking in discord, sure, this week, I do have time in European evenings, something like https://www.timeanddate.com/worldclock/converter.html?iso=20240708T170000&p1=37&p2=202 ?

Reply:
- OK on meta docs.
- I encourage you to try cookieplone for fun. If there is something missing, or is not clear how to do, then file an issue.
- I think an explanation page would be good. I do not know all the parts of CI, what each part does, or how they al work together to fulfill specified tasks.

  In that context, let's change the document title to "Continuous integration". If we keep "essential" or "practices", then we start telling the reader how to do things or what things they should do, which makes it a how-to guide, not an explanation.

  Also if you use Mermaid for diagrams, you could even draw a picture of the [workflow](https://mermaid.js.org/syntax/flowchart.html).

Reply:
@gforcada to get this PR merged, I will remove the CI stuff entirely. It's not correct and should not be published in its current state. The rest of it is good. We can add the CI stuff in another PR.

I'll work on pulling it out of this PR, and moving it back to the coredev directory to preserve the grammar and MyST work I did. I'll do that sometime this week.